com.unity.render-pipelines.core/Runtime/Common/CollectionExtensions.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace UnityEngine.Rendering
+{
+    /// <summary>
+    /// A set of extension methods for collections
+    /// </summary>
+    public static class CollectionExtensions
+    {
+        /// <summary>
+        /// Removes a given count from the back of the given list
+        /// </summary>
+        /// <param name="inputList">The list to be removed</param>
+        /// <param name="elementsToRemoveFromBack">The number of elements to be removed from the back</param>
+        public static void RemoveBack<T>([NotNull] this IList<T> inputList, int elementsToRemoveFromBack)
```

Comment:
For `inputList` it should be `[DisallowNull]`, not `[NotNull]`

Reply:
`elementsToRemoveFromBack` should be a `uint`

Reply:
Can you create 2 methods:
* TryRemoveExactRange: which tries to remove the specified range, and fails if anything is incorrect
* RemoveRange: which tries its best by clamping its argument

It is best to have very simple straightforward API and then syntaxic sugar on top of it, in case the user need full control

Reply:
I will keep the default behaviour, and throw the need exceptions